File: src/main/java/net/onelitefeather/titan/function/RespawnFunction.java
package net.onelitefeather.titan.function;

import net.minestom.server.event.Event;
import net.minestom.server.event.EventNode;
import net.minestom.server.event.player.PlayerRespawnEvent;

public class RespawnFunction {

    private final NavigationFunction navigationFunction;

    private void respawnListener(PlayerRespawnEvent event) {
        this.navigationFunction.setItems(event.getPlayer());
    }

    private RespawnFunction(EventNode<Event> node, NavigationFunction navigationFunction) {
        this.navigationFunction = navigationFunction;
        node.addListener(PlayerRespawnEvent.class, this::respawnListener);
    }

    public static RespawnFunction instance(EventNode<Event> node, NavigationFunction navigationFunction) {
        return new RespawnFunction(node, navigationFunction);
    }

}

File: src/main/kotlin/net/onelitefeather/titan/function/RespawnFunction.kt
package net.onelitefeather.titan.function

import com.google.inject.Inject
import com.google.inject.name.Named
import net.minestom.server.event.Event
import net.minestom.server.event.EventNode
import net.minestom.server.event.player.PlayerRespawnEvent

class RespawnFunction: TitanFunction {

    @Inject
    lateinit var navigationFunction: NavigationFunction

    @Named("titanNode")
    @Inject
    lateinit var eventNode: EventNode<Event>

    private fun respawnListener(event: PlayerRespawnEvent) {
        this.navigationFunction.setItems(event.player)
    }

    override fun initialize() {
        eventNode.addListener(
            PlayerRespawnEvent::class.java,
            this::respawnListener
        )
    }

    override fun terminate() {
    }
}